dask/dataframe/io/parquet/core.py

```diff
@@ -1119,25 +1191,87 @@ def set_index_columns(meta, index, columns, index_in_columns, auto_index_allowed
     return meta, index, columns
 
 
-def aggregate_row_groups(parts, stats, chunksize):
+def aggregate_row_groups(
+    parts, stats, chunksize, split_row_groups, fs, aggregate_files
+):
     if not stats[0].get("file_path_0", None):
         return parts, stats
 
     parts_agg = []
     stats_agg = []
-    chunksize = parse_bytes(chunksize)
+
+    use_row_group_criteria = split_row_groups and int(split_row_groups) > 1
+    use_chunksize_criteria = bool(chunksize)
+    if use_chunksize_criteria:
+        chunksize = parse_bytes(chunksize)
     next_part, next_stat = [parts[0].copy()], stats[0].copy()
     for i in range(1, len(parts)):
         stat, part = stats[i], parts[i]
-        if (stat["file_path_0"] == next_stat["file_path_0"]) and (
-            (next_stat["total_byte_size"] + stat["total_byte_size"]) <= chunksize
+
+        # Criteria #1 for aggregating parts: parts are within the same file
+        same_path = stat["file_path_0"] == next_stat["file_path_0"]
+        multi_path_allowed = False
+
+        if aggregate_files:
+
+            # Criteria #2 for aggregating parts: The part does not include
+            # row-group information, or both parts include the same kind
+            # of row_group aggregation (all None, or all indices)
+            multi_path_allowed = len(part["piece"]) == 1
+            if not (same_path or multi_path_allowed):
+                rgs = set(list(part["piece"][1]) + list(next_part[-1]["piece"][1]))
+                multi_path_allowed = (rgs == {None}) or (None not in rgs)
+
+            # Criteria #3 for aggregating parts: The parts share a
+            # directory at the "depth" allowed by `aggregate_files`
+            if not same_path and multi_path_allowed:
+                if aggregate_files is True:
+                    multi_path_allowed = True
+                elif isinstance(aggregate_files, int):
```

Comment:
I believe this code branch is still missing in the docs. At least I couldn't find a reference to an integer input

Reply:
I just read a bit more code and this might just be the encoded hierarchy depth. if that is true, that's fine. I might suggest to use a different variable name but I am not the best judge here. Feel free to ignore the comments regarding this

Reply:
I actually aggree with you.  The `aggregate_files` variable is trheaded through much of the read_parquet API, because it was originally used to specify a general regex expression.  However, now that we are converting string input to a hierarchy depth, we might as well use a distinct variable name - Perhaps `aggregation depth`?

Reply:
I like the depth much better 👍